src/content/components/IntervalMarkerOverview.js

```diff
@@ -1,44 +1,189 @@
 import React, { Component, PropTypes } from 'react';
 import shallowCompare from 'react-addons-shallow-compare';
+import classNames from 'classnames';
+import { timeCode } from '../../common/time-code';
 import { withSize } from '../with-size';
 
+type MarkerState = 'PRESSED' | 'HOVERED' | 'NONE';
+
 class IntervalMarkerOverview extends Component {
   constructor(props) {
     super(props);
-    this._onSelect = this._onSelect.bind(this);
+    this.state = { hoveredItem: null, mouseDownItem: null };
+    this._onMouseDown = this._onMouseDown.bind(this);
+    this._onMouseMove = this._onMouseMove.bind(this);
+    this._onMouseUp = this._onMouseUp.bind(this);
+    this._onMouseOut = this._onMouseOut.bind(this);
+    this._canvas = null;
+    this._takeCanvasRef = c => { this._canvas = c; };
+  }
+
+  _scheduleDraw() {
+    if (!this._requestedAnimationFrame) {
+      this._requestedAnimationFrame = true;
+      window.requestAnimationFrame(() => {
+        this._requestedAnimationFrame = false;
+        if (this._canvas) {
+          timeCode('IntervalMarkerTimeline render', () => {
+            this.drawCanvas(this._canvas);
+          });
+        }
+      });
+    }
+  }
+
+  _hitTest(e) {
+    const r = this._canvas.getBoundingClientRect();
+    const { width, rangeStart, rangeEnd, intervalMarkers, styles } = this.props;
+    const x = e.pageX - r.left;
+    const y = e.pageY - r.top;
+    const time = rangeStart + x / width * (rangeEnd - rangeStart);
+
+    // Walk the list of intervalMarkers backwards to find the marker under the
+    // mouse. Later markers are drawn on top and we want to find the topmost one.
+    for (let i = intervalMarkers.length - 1; i >= 0; i--) {
+      const { start, dur, name } = intervalMarkers[i];
+      if (time < start || time >= start + dur) {
+        continue;
+      }
+      const style = (name in styles) ? styles[name] : styles.default;
+      if (y >= style.top && y < style.top + style.height) {
+        return intervalMarkers[i];
+      }
+    }
+    return null;
   }
 
-  _onSelect(e) {
-    this.props.onSelect(this.props.threadIndex, +e.target.getAttribute('data-start'), +e.target.getAttribute('data-end'));
+  _onMouseMove(e) {
+    const hoveredItem = this._hitTest(e);
+    this.setState({ hoveredItem });
   }
 
   _onMouseDown(e) {
-    e.stopPropagation();
+    const mouseDownItem = this._hitTest(e);
+    this.setState({ mouseDownItem });
+    if (mouseDownItem !== null) {
+      if (e.target.setCapture) {
```

Comment:
Whoa, I had no idea `setCapture` existed. Nice! Is it not well supported, hence the if?